core/app/components/FormBuilder/ElementPanel/ComponentConfig/index.tsx

```diff
@@ -1,61 +1,40 @@
 "use client";
 
-import type { ComponentType } from "react";
+import type React from "react";
 
 import dynamic from "next/dynamic";
 
 import { InputComponent } from "db/public";
 import { Skeleton } from "ui/skeleton";
 
-import type { ComponentConfigFormProps, InnerFormProps } from "./types.ts";
-
-export const ComponentConfig = ({ component, ...props }: ComponentConfigFormProps) => {
-	let ConfigForm: ComponentType<InnerFormProps>;
-	switch (component) {
-		case InputComponent.checkbox:
-			ConfigForm = dynamic(() => import("./Checkbox.tsx"), {
-				ssr: false,
-				loading: () => <Skeleton className="h-full w-full" />,
-			});
-			break;
-		case InputComponent.confidenceInterval:
-			ConfigForm = dynamic(() => import("./ConfidenceInterval.tsx"), {
-				ssr: false,
-				loading: () => <Skeleton className="h-full w-full" />,
-			});
-			break;
-		case InputComponent.datePicker:
-			ConfigForm = dynamic(() => import("./DatePicker.tsx"), {
-				ssr: false,
-				loading: () => <Skeleton className="h-full w-full" />,
-			});
-			break;
-		case InputComponent.fileUpload:
-			ConfigForm = dynamic(() => import("./FileUpload.tsx"), {
-				ssr: false,
-				loading: () => <Skeleton className="h-full w-full" />,
-			});
-			break;
-		case InputComponent.memberSelect:
-			ConfigForm = dynamic(() => import("./MemberSelect.tsx"), {
-				ssr: false,
-				loading: () => <Skeleton className="h-full w-full" />,
-			});
-			break;
-		case InputComponent.textArea:
-			ConfigForm = dynamic(() => import("./TextArea.tsx"), {
-				ssr: false,
-				loading: () => <Skeleton className="h-full w-full" />,
-			});
-			break;
-		case InputComponent.textInput:
-			ConfigForm = dynamic(() => import("./TextInput.tsx"), {
-				ssr: false,
-				loading: () => <Skeleton className="h-full w-full" />,
-			});
-			break;
-		default:
-			return null;
-	}
-	return <ConfigForm {...props} />;
+import type { ComponentConfigFormProps } from "./types.ts";
+
+const toDynamic = (path: string) =>
+	// this dynamic import path needs to provide enough information for webpack/turbopack
+	// to be able to find it. The relative path and the extension are enough, but something like
+	// `import(path)` will not work.
+	dynamic(() => import(`./${path}.tsx`), {
```

Comment:
honestly I didn't spend that much time trying to get to this version, because of this warning i saw on the next/dynamic docs!
![image](https://github.com/user-attachments/assets/f1f84943-1ff2-4e19-a7fd-967f2e42b03e)

and then i read a few issues where people had various hacks to get around this and I just thought most of them looked like they probably didn't work. thanks for showing me the way!